conn/physic/units_test.go

```diff
@@ -536,3 +538,255 @@ func BenchmarkCelsiusFloatg(b *testing.B) {
+		{"rounding(5.5)", decimal{"55", -1, positive}, 6},
+		{"rounding(5.4)", decimal{"54", -1, positive}, 5},
+		{"rounding(-5.6)", decimal{"56", -1, negative}, -6},
+		{"rounding(-5.5)", decimal{"55", -1, negative}, -6},
+		{"rounding(-5.4)", decimal{"54", -1, negative}, -5},
+		{"rounding(0.6)", decimal{"6", -1, positive}, 1},
+		{"rounding(0.5)", decimal{"5", -1, positive}, 1},
+		{"rounding(0.4)", decimal{"4", -1, positive}, 0},
+		{"rounding(-0.6)", decimal{"6", -1, negative}, -1},
+		{"rounding(-0.5)", decimal{"5", -1, negative}, -1},
+		{"rounding(-0.4)", decimal{"4", -1, negative}, -0},
+	}
+
+	fails := []struct {
+		name     string
+		in       decimal
+		expected int64
+	}{
+		{"max+1", decimal{"9223372036854775808", 0, positive}, 9223372036854775807},
+		{"-max-1", decimal{"9223372036854775808", 0, negative}, -9223372036854775807},
+		{"non digit in decimal.digit)", decimal{"1a", 0, positive}, 0},
+		{"non digit in decimal.digit)", decimal{"2.7b", 0, negative}, 0},
+		{"exponet too large for int64", decimal{"123", 20, positive}, 0},
+		{"exponet too small for int64", decimal{"123", -20, positive}, 0},
+		{"max*10^1", decimal{"9223372036854775807", 1, positive}, 9223372036854775807},
+		{"-max*10^1", decimal{"9223372036854775807", 1, negative}, -9223372036854775807},
+	}
+
+	for _, tt := range succeeds {
+		got, err := dtoi(tt.in, 0)
+
+		if got != tt.expected {
+			t.Errorf("case dtoi() %s got %v expected %v", tt.name, got, tt.expected)
+		}
+		if err != nil {
+			t.Errorf("case dtoi() %s got an unexpected error %v", tt.name, err)
+		}
+	}
+
+	for _, tt := range fails {
+		got, err := dtoi(tt.in, 0)
+
+		if got != tt.expected {
+			t.Errorf("case dtoi() %s got %v expected %v", tt.name, got, tt.expected)
+		}
+		if err == nil {
+			t.Errorf("case dtoi() %s expected %v but got nil", tt.name, err)
+		}
+	}
+}
+
+func TestPrefix(t *testing.T) {
+	tests := []struct {
+		name   string
+		prefix rune
+		want   prefix
+		n      int
+	}{
+		{"pico", 'p', pico, 1},
+		{"nano", 'n', nano, 1},
+		{"micro", 'u', micro, 1},
+		{"mu", 'µ', micro, 2},
+		{"milli", 'm', milli, 1},
+		{"none", 0, none, 0},
+		{"kilo", 'k', kilo, 1},
+		{"mega", 'M', mega, 1},
+		{"giga", 'G', giga, 1},
+		{"tera", 'T', tera, 1},
+	}
+	for _, tt := range tests {
+		got, n := parseSIPrefix(tt.prefix)
+
+		if got != tt.want || n != tt.n {
+			t.Errorf("wanted prefix %d, and len %d, but got prefix %d, and len %d", tt.want, tt.n, got, n)
+		}
+
+	}
+}
+
+func TestParseError(t *testing.T) {
+	tests := []struct {
+		name string
+		err  error
+		want string
+	}{
+		{"empty", &parseError{s: "", err: nil}, "parse error"},
+		{"empty", &parseError{s: "", err: errors.New("test")}, "parse error: test: \"" + "\""},
+		{"noUnits", noUnits("someunit"), "parse error: no units provided, need: \"someunit\""},
+	}
+	for _, tt := range tests {
+		got := tt.err.Error()
+
+		if got != tt.want {
+			t.Errorf("wanted err string:\n%s but got:\n%s", tt.want, got)
+		}
+
```

Comment:
Style nit: I'm not sure why you occasionally add an empty line before a closing bracket.